birdhouse/deployment/deploy.sh

```diff
@@ -178,8 +178,10 @@ set -x
 # restart everything, only changed containers will be destroyed and recreated
 ./pavics-compose.sh up -d
 
-# immediately cache the new notebook image for faster startup by JupyterHub
-docker pull $DOCKER_NOTEBOOK_IMAGE
+# immediately cache the new notebook images for faster startup by JupyterHub
+docker pull $PAVICS_E2E_NOTEBOOK_IMAGE
+docker pull $EO_NOTEBOOK_IMAGE
+docker pull $NLP_NOTEBOOK_IMAGE
```

Comment:
I am not fond of adding all these images directly here.  I think we have to think bigger, that PAVICS will be deployed on many different organizations.  We do not want to force all the orgs to always download all these gigantic images.

The old `DOCKER_NOTEBOOK_IMAGE` var name was meant to be generic so if another org wants, they can simply override the default value in `default.env` and have their own flavor of Jupyter env.

So I would suggest we move these extra jupyter env into an `optional-components/` (https://github.com/bird-house/birdhouse-deploy/tree/master/birdhouse/optional-components).  There are plenty of existing examples you can based upon.  One new thing is this `docker pull` part and you can use the `post-docker-compose-up` script https://github.com/bird-house/birdhouse-deploy/blob/3718a320780bc08cc18632d3c79bb196b2749633/birdhouse/pavics-compose.sh#L193 to do what you need.

Here an example of a `pre-docker-compose-up` script that is equivalent to this `post-docker-compose-up` script: https://github.com/bird-house/birdhouse-deploy/blob/3718a320780bc08cc18632d3c79bb196b2749633/birdhouse/components/monitoring/pre-docker-compose-up#L3-L5